### PR TITLE
fix wandb off-by-one

### PR DESCRIPTION
**Pull request by dreamflasher**
fixes both issues in https://github.com/fastai/fastai/issues/3066

### CODE REVIEW

**Comment by jph00** [comment]
I fixed the smooth_loss issue this morning by setting the callback order. Feel free to reopen this or open a new PR if there's still an unresolved issue.

**Comment by dreamflasher** [comment]
@jph00 I can't reopen, would you be so kind? This fixes two issues – thanks for fixing one, I'd like to adjust this PR so that it fixes the other issue I linked to, thanks!

**Comment by jph00** [comment]
Is this ready now @dreamflasher ?

**Comment by borisdayma** [comment]
Let me do a test.
I'm scared of a side effect which will be that last step is uncommitted in colab until people call `wandb.finish()` (which is typically never done). This is not needed in scripts as it will be automatically called when it finishes.

File: fastai/callback/wandb.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def after_fit(self):
                 log_model(self.save_model.last_saved_path, metadata=metadata)
         self.run = True
         if self.log_preds: self.remove_cb(FetchPredsCallback)
-        wandb.log({}) # ensure sync of last step
+        wandb.log({}, step=self._wandb_step) # ensure sync of last step
```
Comment (borisdayma) [inline fastai/callback/wandb.py:123]: I believe this would be equivalent to just removing that line.

Comment (borisdayma) [reply fastai/callback/wandb.py:123]: Maybe the solution should be to do:
```
# ensure sync of last step
wandb.log({})
self._wandb_step += 1
```

Comment (dreamflasher) [reply fastai/callback/wandb.py:123]: @borisdayma can you run all three variants in your test suite?

Comment (dreamflasher) [reply fastai/callback/wandb.py]: @borisdayma Looks like a reset to upstream, closes an open PR. In any case: I just tested the your last suggestion and this works!

**Pull request closed without merging**